swift_package/BrainFlow/Sources/BoardShim.swift

```diff
@@ -0,0 +1,606 @@
+    }
+
+    /**
+     * get row index in returned by get_board_data() 2d array which contains markers
+     */
+    static func getMarkerChannel (_ boardId: BoardIds, _ preset: BrainFlowPresets = .DEFAULT_PRESET) throws -> Int32
+    {
+        var markerChannel: Int32 = 0
+        let errorCode = get_marker_channel (boardId.rawValue, preset.rawValue, &markerChannel)
+        try checkErrorCode("Error in board info getter", errorCode)
+
+        return markerChannel
+    }
+
+    /**
+     * get number of rows in returned by get_board_data() 2d array
+     */
+    static func getNumRows (_ boardId: BoardIds, _ preset: BrainFlowPresets = .DEFAULT_PRESET) throws -> Int32 {
+        var numRows: Int32 = 0
+        let errorCode = get_num_rows (boardId.rawValue, preset.rawValue, &numRows)
+        try checkErrorCode("Error in board info getter", errorCode)
+
+        return numRows
+    }
+
+    /**
+     * get row index in returned by get_board_data() 2d array which contains package nums
+     */
+    static func getPackageNumChannel (_ boardId: BoardIds, _ preset: BrainFlowPresets = .DEFAULT_PRESET) throws -> Int32 {
+        var pkgNumChannel: Int32 = 0
+        let errorCode = get_package_num_channel (boardId.rawValue, preset.rawValue, &pkgNumChannel)
+        try checkErrorCode("Error in board info getter", errorCode)
+
+        return pkgNumChannel
+    }
+
+    /**
+     * get row index in returned by get_board_data() 2d array which contains battery level
+     */
+    static func getBatteryChannel (_ boardId: BoardIds, _ preset: BrainFlowPresets = .DEFAULT_PRESET) throws -> Int32 {
+        var batteryChannel: Int32 = 0
+        let errorCode = get_battery_channel (boardId.rawValue, preset.rawValue, &batteryChannel)
+        try checkErrorCode("Error in board info getter", errorCode)
+
+        return batteryChannel
+    }
+
+    /**
+     * Get names of EEG electrodes in 10-20 system. Only if electrodes have freezed
+     * locations
+     */
+    static func getEEGnames (_ boardId: BoardIds, _ preset: BrainFlowPresets = .DEFAULT_PRESET) throws -> [String] {
+        var stringLen: Int32 = 0
+        var bytes = [CChar](repeating: 0, count: 4096)
+        let errorCode = get_eeg_names (boardId.rawValue, preset.rawValue, &bytes, &stringLen)
+        try checkErrorCode("Error in board info getter", errorCode)
+        let EEGnames = bytes.toString(stringLen)
+
+        return EEGnames.components(separatedBy: ",")
+    }
+
+    /**
+     * Get board description
+     */
+    static func getBoardDescr (_ boardId: BoardIds, _ preset: BrainFlowPresets = .DEFAULT_PRESET) throws -> BoardDescription {
+        var boardDescrStr = [CChar](repeating: CChar(0), count: 16000)
+        var stringLen: Int32 = 0
+        let errorCode = get_board_descr (boardId.rawValue, preset.rawValue, &boardDescrStr, &stringLen)
+        try checkErrorCode("failed to get board info", errorCode)
+
+        return try BoardDescription(boardDescrStr.toString(stringLen))
```

Comment:
to/from json static methods would be better

Reply:
updated